src/py21cmfast/wrapper/inputs.py

```diff
@@ -1492,6 +1568,13 @@ def asdict(
                 if isinstance(getattr(self, k), InputStruct)
             }
 
+        if only_cstruct_params:
+            for k, v in dct.items():
+                kls = getattr(self, k)
+                if isinstance(kls, InputStruct):
+                    dct[k] = {kk: vv for kk, vv in v.items() if kk in kls.cdict} | {
+                        kk: getattr(kls, kk) for kk in kls.cdict if kk not in dct[k]
+                    }
```

Comment:
Two quick things here:

1: maybe I'm wrong about the convention, but `kls` makes me think it's the class itself, when it's an instance.
2: Is the RHS of the dictionary merge for the new non-field properties? Could a user mess these up e.g: `inputs.HII_DIM=20`, `inputs.DIM=40`, `inputs.HIRES_TO_LOWRES_FACTOR=10` and then propogate them with a clone?

Reply:
What I meant by 3 is similar to Jordan's comment that i don't see a strict requirement for the dimensions and ratio parameters to be consistent. If someone uses a template with `BOX_LEN` and `DIM` and then evolves it with `HIRES_TO_LOWRES_FACTOR` or `LOWRES_CELL_SIZE_MPC`, their changes would be ignored without warning or error. I'm not sure the best way to address this, I do like the precenence of the old parameters, but maybe we should throw an error when evolving one of the dimension ratios *after* the dimension parameters are set?